pkgs/build-support/dotnet/build-dotnet-module/default.nix

```
@@ -135,43 +151,78 @@ in stdenvNoCC.mkDerivation (args // {
     inherit nuget-source;
 
     fetch-deps = let
-      exclusions = dotnet-sdk.passthru.packages { fetchNuGet = attrs: attrs.pname; };
+      # Because this list is rather long its put in its own store path to maintain readability of the generated script
+      exclusions = writeText "nuget-package-exclusions" (lib.concatStringsSep "\n" (dotnet-sdk.passthru.packages { fetchNuGet = attrs: attrs.pname; }));
+
+      # Used to restore dependencies for all platforms supported by the package
+      systemToDotnetRid = {
+        "x86_64-linux" = "linux-x64";
+        "aarch64-linux" = "linux-arm64";
+        "x86_64-darwin" = "osx-x64";
+        "aarch64-darwin" = "osx-arm64";
+      };
```

Lets add an throw unsupported system just to be safe.